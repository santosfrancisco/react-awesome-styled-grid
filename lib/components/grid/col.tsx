import styled, { css } from "styled-components";
import config, { DIMENSIONS } from "../../config";
import { GridColProps } from "@site/types";

export const Col = styled.div<GridColProps>`
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;
  flex-direction: column;

  ${(props) =>
    !props.noGutter &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
      padding-right: ${config(props).gutterWidth[dimension] / 2}rem;
      padding-left: ${config(props).gutterWidth[dimension] / 2}rem;
    `
      )}
    `}

  ${(props) => css`
    ${DIMENSIONS.map(
      (dimension) =>
        props[dimension] &&
        config(props).breakpoints[dimension] &&
        config(props).media[dimension]`
      flex: 1 1 ${(+props[dimension] / config(props).columns[dimension]) * 100}%;
      max-width: ${(+props[dimension] / config(props).columns[dimension]) * 100}%;
    `
    )}
  `}

  ${(props) =>
    props.offset &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
    ${
      typeof props.offset === "object"
        ? props.offset[dimension] !== undefined &&
          `margin-left: ${props.offset[dimension] > 0 ? (props.offset[dimension] / config(props).columns[dimension]) * 100 : 0}%`
        : `margin-left: ${(props.offset / config(props).columns["xs"]) * 100}%`
    };
    `
      )}
    `}

  ${(props) =>
    props.order !== undefined &&
    css`
      ${DIMENSIONS.map(
        (dimension) =>
          config(props).breakpoints[dimension] &&
          config(props).media[dimension]`
    ${
      typeof props.order === "object"
        ? props.order[dimension] !== undefined &&
          `order: ${props.order[dimension]}`
        : `order: ${props.order}`
    };
    `
      )}
    `}

  ${(props) =>
    props.reverse &&
    css`
      ${Array.isArray(props.reverse)
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`
        flex-direction: ${Array.isArray(props.reverse) && props.reverse.indexOf(dimension) !== -1 ? "column-reverse" : "column"};
      `
          )
        : "flex-direction: column-reverse;"}
    `}

  ${(props) =>
    props.align &&
    css`
      ${typeof props.align === "object"
        ? DIMENSIONS.map(
            (dimensions) =>
              config(props).breakpoints[dimensions] &&
              config(props).media[
                dimensions
              ]`${props.align[dimensions] && `align-items: ${props.align[dimensions]}`}`
          )
        : `align-items: ${props.align};`}
    `}
  
  ${(props) =>
    props.justify &&
    css`
      ${typeof props.justify === "object"
        ? DIMENSIONS.map(
            (dimensions) =>
              config(props).breakpoints[dimensions] &&
              config(props).media[
                dimensions
              ]`${props.justify[dimensions] && `justify-content: ${props.justify[dimensions]}`}`
          )
        : `justify-content: ${props.justify};`}
    `}
  
  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
