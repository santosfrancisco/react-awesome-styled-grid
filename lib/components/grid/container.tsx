import styled, { css } from "styled-components";
import isPropValid from "@emotion/is-prop-valid";
import config, { DIMENSIONS } from "../../config";
import { GridContainerProps } from "../../types";

export const Container = styled("div").withConfig({
  shouldForwardProp: (prop) => isPropValid(prop)
})<GridContainerProps>`
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;

  ${(props) => css`
    ${DIMENSIONS.map(
      (d) =>
        config(props).container[d] &&
        config(props).media[d]`
      padding-left: ${config(props).paddingWidth[d]}rem;
      padding-right: ${config(props).paddingWidth[d]}rem;
    `
    )}
  `}

  ${(props) =>
    !props.fluid &&
    css`
      ${DIMENSIONS.map(
        (d) =>
          config(props).container[d] &&
          config(props).media[d]`
      ${
        typeof config(props).container[d] === "number"
          ? `max-width: ${config(props).container[d]}rem;`
          : `width: 100%;`
      }
    `
      )}
    `}

  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
