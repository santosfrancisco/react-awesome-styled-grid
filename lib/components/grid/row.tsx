import styled, { css } from "styled-components";
import config, { DIMENSIONS } from "../../config";
import { GridRowProps } from "@site/types";

export const Row = styled.div<GridRowProps>`
  box-sizing: border-box;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: wrap;

  ${(props) => css`
    ${DIMENSIONS.map(
      (dimension) =>
        config(props).container[dimension] &&
        config(props).media[dimension]`
      margin-left: -${config(props).gutterWidth[dimension] / 2}rem;
      margin-right: -${config(props).gutterWidth[dimension] / 2}rem;
    `
    )}
  `}

  ${(props) =>
    props.reverse &&
    css`
      ${Array.isArray(props.reverse)
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`
         ${
           Array.isArray(props.reverse) &&
           props.reverse.indexOf(dimension) !== -1
             ? `
             flex-direction: row-reverse;
             flex-wrap: wrap-reverse;
             `
             : `
                 flex-direction: row;
                 flex-wrap: wrap;
               `
         };
        
      `
          )
        : `
            flex-direction: row-reverse;
            flex-wrap: wrap-reverse;
          `}
    `}

  ${(props) =>
    props.align &&
    css`
      ${typeof props.align === "object"
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[
                dimension
              ]`${props.align[dimension] && `align-items: ${props.align[dimension]}`}`
          )
        : `align-items: ${props.align};`}
    `}
  
  ${(props) =>
    props.justify &&
    css`
      ${typeof props.justify === "object"
        ? DIMENSIONS.map(
            (dimension) =>
              config(props).breakpoints[dimension] &&
              config(props).media[dimension]`${
                props.justify[dimension] &&
                `justify-content: ${props.justify[dimension]}`
              }`
          )
        : `justify-content: ${props.justify};`}
    `}

  ${({ debug }) =>
    debug &&
    css`
      background-color: #5901ad40;
      outline: #fff solid 1px;
    `}
`;
