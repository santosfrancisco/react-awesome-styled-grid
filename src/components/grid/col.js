import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import config, { DIMENSIONS } from '../../config'

const Col = styled.div`
  box-sizing: border-box;
  flex: 1 0 auto;
  max-width: 100%;
  display: flex;

  ${p => !p.noGutter && css`
    ${DIMENSIONS.map(d =>
    config(p).breakpoints[d] && config(p).media[d]`
      padding-right: ${config(p).gutterWidth[d] / 2}rem;
      padding-left: ${config(p).gutterWidth[d] / 2}rem;
    `)}
  `}

  ${p => css`
    ${DIMENSIONS.map(d =>
    config(p).breakpoints[d] && config(p).media[d]`
      ${p[d] && `
        flex: 1 1 ${(p[d] / config(p).columns[d]) * 100}%;
        max-width: ${(p[d] / config(p).columns[d]) * 100}%;
      `}
    `)}
  `}

  ${p => p.offset && css`
    ${DIMENSIONS.map(d =>
    config(p).breakpoints[d] && config(p).media[d]`
    ${typeof p.offset === 'object'
    ? p.offset[d] !== undefined && `margin-left: ${p.offset[d] > 0 ? (p.offset[d] / config(p).columns[d]) * 100 : 0}%`
    : `margin-left: ${(p.offset / config(p).columns['xs']) * 100}%`};
    `)}
  `}

  ${p => p.reverse && css`
    ${Array.isArray(p.reverse)
    ? DIMENSIONS.map(d => config(p).breakpoints[d] && config(p).media[d]`
        flex-direction: ${p.reverse.indexOf(d) !== -1 ? 'column-reverse' : 'column'};
      `)
    : 'flex-direction: column-reverse;'}
  `}

  ${p => p.align && css`
    ${typeof p.align === 'object'
    ? DIMENSIONS.map(d => config(p).breakpoints[d] && config(p).media[d]`${p.align[d] && `align-items: ${p.align[d]}`}`)
    : `align-items: ${p.align};`}
  `}
  
  ${p => p.justify && css`
    ${typeof p.justify === 'object'
    ? DIMENSIONS.map(d => config(p).breakpoints[d] && config(p).media[d]`${p.justify[d] && `justify-content: ${p.justify[d]}`}`)
    : `justify-content: ${p.justify};`}
  `}
  
  ${({ debug }) => debug && css`
    background-color: #5901ad40;
    outline: #fff solid 1px;
  `}
`

Col.displayName = 'Col'

const numberOrString = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number
])

const stringOrObject = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.object
])

const numberOrObject = PropTypes.oneOfType([
  PropTypes.bool,
  PropTypes.object
])

const boolOrArray = PropTypes.oneOfType([
  PropTypes.bool,
  PropTypes.array
])

Col.propTypes = {
  xs: numberOrString,
  sm: numberOrString,
  md: numberOrString,
  lg: numberOrString,
  xl: numberOrString,
  align: stringOrObject,
  justify: stringOrObject,
  offSet: numberOrObject,
  reverse: boolOrArray,
  noGutter: PropTypes.bool,
  children: PropTypes.node,
  debug: PropTypes.bool
}

export default Col
