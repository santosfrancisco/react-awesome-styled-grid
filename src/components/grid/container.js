import PropTypes from 'prop-types'
import styled, {
  css
} from 'styled-components'

import config, {
  DIMENSIONS
} from '../../config'

const Container = styled.div`
  margin-right: auto;
  margin-left: auto;

  ${p => css`
    ${DIMENSIONS.map(d =>
    config(p).container[ d ] && config(p).media[ d ]`
      padding-left: ${config(p).paddingWidth[ d ]}rem;
      padding-right: ${config(p).paddingWidth[ d ]}rem;
    `)}
  `}
  

  ${p => !p.fluid && css`
    ${DIMENSIONS.map(d =>
    config(p).container[ d ] && config(p).media[ d ]`
      ${typeof config(p).container[d] === 'number' ? `max-width: ${config(p).container[ d ]}rem;` : `max-width: 100%;`}
    `)}
  `}

  ${({ debug }) => debug && css`
    background-color: #5901ad40;
    outline: #fff solid 1px;
  `}
`

Container.displayName = 'Container'

Container.propTypes = {
  fluid: PropTypes.bool,
  children: PropTypes.node,
  debug: PropTypes.bool
}

export default Container
