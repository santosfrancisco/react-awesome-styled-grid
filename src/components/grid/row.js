
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import config, { DIMENSIONS } from '../../config'

const Row = styled.div`
  box-sizing: border-box;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: wrap;
  
  ${p => css`
    ${DIMENSIONS.map(d =>
    config(p).container[ d ] && config(p).media[ d ]`
      margin-left: -${config(p).gutterWidth[d] / 2}rem;
      margin-right: -${config(p).gutterWidth[d] / 2}rem;
    `)}
  `}

  ${p => p.reverse && css`
    ${Array.isArray(p.reverse)
    ? DIMENSIONS.map(d =>
      config(p).breakpoints[d] && config(p).media[d]`
        flex-direction: ${p.reverse.indexOf(d) !== -1 ? `row-reverse` : `row`};
      `)
    : 'flex-direction: row-reverse;'}
  `}

  ${p => p.align && css`
    ${typeof p.align === 'object'
    ? DIMENSIONS.map(d => config(p).breakpoints[d] && config(p).media[d]`${p.align[d] && `align-items: ${p.align[d]}`}`)
    : `align-items: ${p.align};`}
  `}
  
  ${p => p.justify && css`
    ${typeof p.justify === 'object'
    ? DIMENSIONS.map(d => config(p).breakpoints[d] && config(p).media[d]`${p.justify[d] && `justify-content: ${p.justify[d]}`}`)
    : `justify-content: ${p.justify};`}
  `}

  ${({ debug }) => debug && css`
    background-color: #5901ad40;
    outline: #fff solid 1px;
  `}
`

Row.displayName = 'Row'

const boolOrArray = PropTypes.oneOfType([
  PropTypes.bool,
  PropTypes.array
])

const stringOrObject = PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.object
])

Row.propTypes = {
  reverse: boolOrArray,
  align: stringOrObject,
  justify: stringOrObject,
  children: PropTypes.node,
  debug: PropTypes.bool
}

export default Row
