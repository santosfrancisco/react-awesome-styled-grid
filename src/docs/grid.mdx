---
name: Usage
route: /usage
---

import { Container, Row, Col, Visible, Hidden, ScreenClass, ScreenBadge } from '../'

# Usage

```js
import { Container, Row, Col, Visible, Hidden, ScreenClass, ScreenBadge } from '../'
```

Resizes your window to see the responsive magic happening ðŸ˜Ž

## Responsive

<Container>
  <Row>
    <Col debug xs={2} sm={6} md={2} lg={6}>
      xs={2} sm={6} md={2} lg={6}
    </Col>
    <Col debug xs={2} sm={2} md={6} lg={6}>
      xs={2} sm={2} md={6} lg={6}
    </Col>
  </Row>
  <Row>
    <Col debug xs={2} sm={2} md={6} lg={6}>
      xs={2} sm={2} md={6} lg={6}
    </Col>
    <Col debug xs={2} sm={6} md={2} lg={6}>
      xs={2} sm={6} md={2} lg={6}
    </Col>
  </Row>
  <Row>
    <Col debug xs={2} sm={6} md={2} lg={6}>
      xs={2} sm={6} md={2} lg={6}
    </Col>
    <Col debug xs={2} sm={2} md={6} lg={6}>
      xs={2} sm={2} md={6} lg={6}
    </Col>
  </Row>
</Container>

> ðŸ’¡**Tip!** if no dimension is defined, the column size will be **auto**. 
(i.e. md={6}. This means that ```xs``` and ```sm``` is auto, 
and from ```md```, the size will be 6 columns)

```js
<Container>
  <Row>
    <Col debug xs={2} sm={6} md={2} lg={6}>
      xs={2} sm={6} md={2} lg={6}
    </Col>
    <Col debug xs={2} sm={2} md={6} lg={6}>
      xs={2} sm={2} md={6} lg={6}
    </Col>
  </Row>
  <Row>
    <Col debug xs={1} sm={4} md={3} lg={4}>
      xs={1} sm={4} md={3} lg={4}
    </Col>
    <Col debug xs={3} sm={4} md={5} lg={8}>
      xs={3} sm={4} md={5} lg={8}
    </Col>
  </Row>
</Container>
```

## Offset

### Using number

if you enter a number, the number of columns of the screen xs will be used as base. in this example, 2 columns of 4 will be used for both offset and column size, ie 50%.

<Container>
  <Row>
    <Col debug xs={2} offset={2}>50% offset</Col>
  </Row>
</Container>

### Using object

in this case, the ```offset``` prop must receive an object with the screens and their respective values.

<Container>
  <Row><Col debug xs={1} sm={1} lg={1} offset={{ xs:3, sm:7, lg:11}}>&nbsp;</Col></Row>
  <Row><Col debug xs={1} sm={1} lg={2} offset={{ xs:3, sm:7, lg:10}}>&nbsp;</Col></Row>
  <Row><Col debug xs={1} sm={2} lg={3} offset={{ xs:3, sm:6, lg:9}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={3} lg={4} offset={{ xs:2, sm:5, lg:8}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={4} lg={5} offset={{ xs:2, sm:4, lg:7}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={5} lg={6} offset={{ xs:2, sm:3, lg:6}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={5} lg={7} offset={{ xs:1, sm:3, lg:5}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={5} lg={8} offset={{ xs:1, sm:3, lg:4}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={6} lg={9} offset={{ xs:1, sm:2, lg:3}}>&nbsp;</Col></Row>
  <Row><Col debug xs={4} sm={7} lg={10} offset={{ xs:0, sm:1, lg:2}}>&nbsp;</Col></Row>
  <Row><Col debug xs={4} sm={8} lg={11} offset={{ xs:0, sm:0, lg:1}}>&nbsp;</Col></Row>
</Container>

```js
<Container>
  <Row><Col debug xs={1} sm={1} lg={1} offset={{ xs:3, sm:7, lg:11}}>&nbsp;</Col></Row>
  <Row><Col debug xs={1} sm={1} lg={2} offset={{ xs:3, sm:7, lg:10}}>&nbsp;</Col></Row>
  <Row><Col debug xs={1} sm={2} lg={3} offset={{ xs:3, sm:6, lg:9}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={3} lg={4} offset={{ xs:2, sm:5, lg:8}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={4} lg={5} offset={{ xs:2, sm:4, lg:7}}>&nbsp;</Col></Row>
  <Row><Col debug xs={2} sm={5} lg={6} offset={{ xs:2, sm:3, lg:6}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={5} lg={7} offset={{ xs:1, sm:3, lg:5}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={5} lg={8} offset={{ xs:1, sm:3, lg:4}}>&nbsp;</Col></Row>
  <Row><Col debug xs={3} sm={6} lg={9} offset={{ xs:1, sm:2, lg:3}}>&nbsp;</Col></Row>
  <Row><Col debug xs={4} sm={7} lg={10} offset={{ xs:0, sm:1, lg:2}}>&nbsp;</Col></Row>
  <Row><Col debug xs={4} sm={8} lg={11} offset={{ xs:0, sm:0, lg:1}}>&nbsp;</Col></Row>
</Container>
```
## Row reverse

```reverse``` receives an array with the screens that will have the flow reversed. if it is passed as boolean, the stream will be inverted at all breakpoints

<Container>
  <Row reverse={['md']}>
    <Col debug xs={2}>Column A</Col>
    <Col debug xs={2}>Column B</Col>
  </Row>
  <Row reverse>
    <Col debug xs={2}>Column A</Col>
    <Col debug xs={2}>Column B</Col>
  </Row>
</Container>

```js
<Container>
  <Row reverse={['md']}>
    <Col debug xs={2}>Column A</Col>
    <Col debug xs={2}>Column B</Col>
  </Row>
  <Row reverse>
    <Col debug xs={2}>Column A</Col>
    <Col debug xs={2}>Column B</Col>
  </Row>
</Container>
```

## Column reverse

```reverse``` receives an array with the screens that will have the flow reversed. if it is passed as boolean, the stream will be inverted at all breakpoints

<Container>
  <Row>
    <Col noGutter reverse={['md', 'lg']}>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column A</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column B</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column C</div>
    </Col>
        <Col noGutter reverse>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column A</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column B</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column C</div>
    </Col>
  </Row>
</Container>

```js
<Container>
  <Row>
    <Col noGutter reverse={['md', 'lg']}>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column A</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column B</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column C</div>
    </Col>
        <Col noGutter reverse>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column A</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column B</div>
      <div style={{backgroundColor: '#5901ad40', outline: '1px solid #fff'}}>Column C</div>
    </Col>
  </Row>
</Container>
```

## Visible

Show a element only in a specific screen size

### Example using only text

<div>
  <Visible xs>This text is visible only on: XS</Visible>
  <Visible sm>This text is visible only on: SM</Visible>
  <Visible md>This text is visible only on: MD</Visible>
  <Visible lg>This text is visible only on: LG</Visible>
  <Visible xl>This text is visible only on: XL</Visible>
</div>

```js
<div>
  <Visible xs>This text is visible only on: XS</Visible>
  <Visible sm>This text is visible only on: SM</Visible>
  <Visible md>This text is visible only on: MD</Visible>
  <Visible lg>This text is visible only on: LG</Visible>
  <Visible xl>This text is visible only on: XL</Visible>
</div>
```

### Example using an Row with Col

<Container>
  <Row>
    <Visible xs sm>
      <Col debug>Visible only on XS and SM screens</Col>
    </Visible>
    <Visible md lg xl>
      <Col debug>Visible only on MD, LG and XL screens</Col>
    </Visible>
  </Row>
</Container>

```js
<Container>
  <Row>
    <Visible xs sm>
      <Col debug>Visible only on XS and SM screens</Col>
    </Visible>
    <Visible md lg xl>
      <Col debug>Visible only on MD, LG and XL screens</Col>
    </Visible>
  </Row>
</Container>
```

## Hidden

Hides a element only in a specific screen size

### Example using only text

<div>
  <Hidden xs><div>This text is hidden only on: XS</div></Hidden>
  <Hidden sm><div>This text is hidden only on: SM</div></Hidden>
  <Hidden md><div>This text is hidden only on: MD</div></Hidden>
  <Hidden lg><div>This text is hidden only on: LG</div></Hidden>
  <Hidden xl><div>This text is hidden only on: XL</div></Hidden>
</div>

```js
<div>
  <Hidden xs><div>This text is hidden only on: XS</div></Hidden>
  <Hidden sm><div>This text is hidden only on: SM</div></Hidden>
  <Hidden md><div>This text is hidden only on: MD</div></Hidden>
  <Hidden lg><div>This text is hidden only on: LG</div></Hidden>
  <Hidden xl><div>This text is hidden only on: XL</div></Hidden>
</div>
```

### Example using an Row with Col

<Container>
  <Row>
    <Hidden xs sm>
      <Col debug>Hidden only on XS and SM screens</Col>
    </Hidden>
    <Hidden md lg xl>
      <Col debug>Hidden only on MD, LG and XL screens</Col>
    </Hidden>
  </Row>
</Container>

```js
<Container>
  <Row>
    <Hidden xs sm>
      <Col debug>Hidden only on XS and SM screens</Col>
    </Hidden>
    <Hidden md lg xl>
      <Col debug>Hidden only on MD, LG and XL screens</Col>
    </Hidden>
  </Row>
</Container>
```


## ScreenClass

ScreenClass receive a prop named `render` and renders this by passing the current screen as parameter

<Container>
  <Row>
    <Col debug><ScreenClass render={(screen) => (
        <p style={{ color: ['sm', 'md'].includes(screen) ? 'red' : 'purple' }}>Hello! Actual screen is {screen}</p>
      )} />
    </Col>
  </Row>
</Container>

```js
<Container>
  <Row>
    <Col debug><ScreenClass render={(screen) => (
        <p style={{ color: ['sm', 'md'].includes(screen) ? 'red' : 'purple' }}>Hello! Actual screen is {screen}</p>
      )} />
    </Col>
  </Row>
</Container>
```

## ScreenBadge

Displays a badge in the lower right corner with the current screen for debug

<ScreenBadge />

```js
  <ScreenBadge />
```