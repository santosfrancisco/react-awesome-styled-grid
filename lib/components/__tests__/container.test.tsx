import { render } from "@testing-library/react";
import { Container } from "../grid/container";

describe("<Container />", () => {
  it("should render default style correctly", () => {
    const { asFragment } = render(<Container>children</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have a max-width 100% when it is a fluid container", () => {
    const { asFragment } = render(<Container fluid>children</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have different style when it`s debug props is true", () => {
    const { asFragment } = render(<Container debug>children</Container>);
    expect(asFragment()).toMatchSnapshot();
  });
});
