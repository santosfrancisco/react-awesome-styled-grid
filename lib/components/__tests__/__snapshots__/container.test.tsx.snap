// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should have a max-width 100% when it is a fluid container 1`] = `
<DocumentFragment>
  .c0 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Container /> should have different style when it\`s debug props is true 1`] = `
<DocumentFragment>
  .c0 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;
  background-color: #5901ad40;
  outline: #fff solid 1px;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    max-width: 90rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    max-width: 90rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`<Container /> should render default style correctly 1`] = `
<DocumentFragment>
  .c0 {
  margin-right: auto;
  margin-left: auto;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;
}

@media only screen and (min-width: 1rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media only screen and (min-width: 1rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 48rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 64rem) {
  .c0 {
    width: 100%;
  }
}

@media only screen and (min-width: 90rem) {
  .c0 {
    max-width: 90rem;
  }
}

@media only screen and (min-width: 120rem) {
  .c0 {
    max-width: 90rem;
  }
}

<div
    class="c0"
  >
    children
  </div>
</DocumentFragment>
`;
